### ISSUE TITLE
streamdeck module can't control Stream Deck if official Stream Deck software running

### ISSUE THREAD

**Issue by jeffisfast** [Hammerspoon/hammerspoon#3032]
I've written a stream deck blanker program that automatically turns off the Stream Deck when my Mac goes to sleep so that the light from the Stream Deck wouldn't illuminate my room.  I'm still using the Elgato official Steam Deck software to configure my buttons.  

While the Elgato software is running, Hammerspoon is unable to control the Steam Deck.  Any suggestions for how I can get around this?  I've tried changing the order the programs start to no avail.

Here's my code: [jeffisfast/my-hammerspoon](https://github.com/jeffisfast/my-hammerspoon)

Thanks

**Comment by latenitefilms** [comment]
You can't run the official Stream Deck app at the same time as the Hammerspoon Stream Deck extension. The only thing I can think of is if you kill the official app when you're computers going to sleep, and then re-launch it when your system wakes up?

**Comment by cmsj** [comment]
@latenitefilms 's suggestion ought to work, but you'll also need to be careful to fully deallocate the `hs.streamdeck` object for the deck (ie set the object to `nil` and call `collectgarbage()`) so the Elgato software can start up.

FWIW the reason the two things can't both talk to the deck is because each of them are essentially USB drivers and there's no way to share a USB device between two drivers.

**Comment by jeffisfast** [comment]
Thanks @latenitefilms and @cmsj .  Understood on the reason they can't both run simultaneously and like the suggestion of quitting the Elgato software first.  I've got my script properly quitting the app and starting it when the machine sleeps or wakes, but have one issue that I'm hoping you have an opinion on.  

It seems that no matter what order I start Hammerspoon or the Elgato software, once the Elgato software is running it seizes the Steam Deck device.  After I quite the Elgato software, I can't seem to get Hammerspoon to reconnect to the Stream Deck.  Right now I'm starting init method when Hammerspoon launches and it finds the StreamDeck whether Elgato is running or not.  If Elgato is not running I can control the Stream Deck.  Once Elgato is running, that connection no longer works, even after I quit Elgato.  I've tried using the .getdevice method to force a reconnection but that doesn't work.  

What is the right way to get it to reconnect to the device?  Many thanks!
